spec/models/collection_spec.rb

```diff
@@ -72,4 +72,44 @@
       end # moderated_status loop
     end # challenges type loop
   end
+
+  describe "all_bookmarked_items_count" do
+    let(:collection) { create(:collection) }
+
+    it "does not include bookmarks of deleted works" do
+      work = create(:posted_work)
+      create(:bookmark, collections: [collection], bookmarkable: work)
+      expect do
+        work.destroy
+      end.to change { collection.all_bookmarked_items_count }.from(1).to(0)
```

Comment:
This expect is cool, I'm gonna start using it.

Reply:
Looks useful. I might adopt that too.